rust/xtask/src/main.rs

```diff
@@ -0,0 +1,79 @@
+use std::{env, path::PathBuf};
+
+mod tasks {
+    use std::{fs::File, io::Write};
+
+    use agama_cli::Cli;
+    use clap::CommandFactory;
+    use clap_complete::aot;
+    use clap_markdown::MarkdownOptions;
+
+    use crate::create_output_dir;
+
+    /// Generate auto-completion code for common shells.
+    pub fn generate_complete() -> std::io::Result<()> {
+        let out_dir = create_output_dir("shell")?;
+
+        let mut cmd = Cli::command();
+        clap_complete::generate_to(aot::Bash, &mut cmd, "agama", &out_dir)?;
+        clap_complete::generate_to(aot::Fish, &mut cmd, "agama", &out_dir)?;
+        clap_complete::generate_to(aot::Zsh, &mut cmd, "agama", &out_dir)?;
```

Comment:
we should probably package each completion package separately with own supplements so on live cd it will install only shell ones.